server/src/main/java/org/opensearch/tracing/SpanHolder.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tracing;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * Mutable wrapper class to store {@link Span}.
+ */
+public class SpanHolder {
```

Comment:
why do we need this class?

Reply:
Transient headers does not allow to update the value for the same key (CURRENT_SPAN). However we need to keep updating the current span whenever any new span get created. Hence as a workaround we have created this wrapper class to update the value of current span by updating the value of the this wrapper instead of updating the key's value in the (transient header) map

Reply:
You can have the map use AtomicReference for Span and then modify the AtomicReference directly instead?

Reply:
Done